engine/src/main/java/org/terasology/world/zones/ZoneRegionFunction.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.world.zones;
+
+import org.terasology.math.geom.BaseVector3i;
+import org.terasology.world.generation.Region;
+
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+public interface ZoneRegionFunction {
+
+    /**
+     * Calculates whether or not the given block is part of this layer.
+     *
+     * @param x the x position to check
+     * @param y the y position to check
+     * @param z the z position to check
+     * @param region the Region in the area
+     * @param zone the zone that contains this function
+     * @return true if the position is within this layer, false otherwise
+     */
+    boolean apply(int x, int y, int z, Region region, Zone zone);
```

Comment:
I don't think this method needs to be overloaded; it's only used in the `Zone`, where I think I swapped using a `Vector3i` for `int`s because the profiler showed a not-insignificant amount of time spent creating the vectors (this method gets called a lot).

There are alternative versions available in the constructors for the `Zone` which use vectors (though I'm not sure of the performance implications of that; I was mainly testing with the layered zones, which don't use the vector methods).